src/Processors/Formats/Impl/PrettyBlockOutputFormat.cpp

```diff
@@ -345,7 +345,7 @@ void PrettyBlockOutputFormat::writeChunk(const Chunk & chunk, PortKind port_kind
         writeReadableNumberTip(chunk);
         writeCString("\n", out);
 
-        if (has_transferred_row)
+        if (has_transferred_row && format_settings.pretty.preserve_border_for_multiline_string)
```

Comment:
In this case, the check is unnecessary because in the [check above](https://github.com/ClickHouse/ClickHouse/pull/63479/files#diff-b9c946c844ee8c90f75bab54aebecc2223032812634841a629913b0396469f3cR336), it will not go into the `splitValueAtBreakLine` function and `transferred_row[j].empty()` will always be true